apps/browser/components/CurrentWorkspace.tsx

```diff
@@ -0,0 +1,110 @@
+import React, {
+  useEffect,
+  useState,
+  PropsWithChildren,
+  MouseEvent,
+} from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faUpload } from "@fortawesome/free-solid-svg-icons";
+import { useDropzone } from "react-dropzone";
+import { readFile } from "@/lib/sys/file";
+import { InMemoryFile } from "@nerfzael/memory-fs";
+import { downloadFilesAsZip } from "@/lib/sys/file/downloadFilesAsZip";
+import clsx from "clsx";
+import IconButton from "./IconButton";
+import FileIcon from "./FileIcon";
+
+interface UploadProps {
+  className?: string;
+  userFiles: InMemoryFile[];
+  onUploadFiles: (files: InMemoryFile[]) => void;
+}
+
+function CurrentWorkspace({
+  userFiles,
+  onUploadFiles,
+  ...props
+}: PropsWithChildren<UploadProps>) {
+  const { className } = props;
+  const [showUpload, setShowUpload] = useState(false);
+  const { acceptedFiles, getRootProps, getInputProps, isDragAccept, open } =
+    useDropzone({ noClick: true });
+
+  function downloadUserFiles(event: MouseEvent<HTMLButtonElement>) {
+    event.preventDefault();
```

Comment:
@cbrzn I ran into a small issue with my rework of this section where the download button also triggers the `open` function. I figured adding this would prevent that from happening, but unfortunately did not resolve the issue. Also, I had removed some of the conditions `showUpload` since the UX is a bit different, but this commit could benefit from a close look in case I missed something when making the revisions.